src/topology/algebra/continuous_functions.lean

```diff
@@ -0,0 +1,42 @@
+-- Copyright (c) 2019 Scott Morrison. All rights reserved.
+-- Released under Apache 2.0 license as described in the file LICENSE.
+-- Authors: Scott Morrison
+
+import topology.basic
+import topology.algebra.ring
+import ring_theory.subring
+
+universes u v
+
+local attribute [elab_simple] continuous.comp
+instance continuous_submonoid (α : Type u) (β : Type v) [topological_space α] [topological_space β]
+  [monoid β] [topological_monoid β] : is_submonoid { f : α → β | continuous f } :=
+{ one_mem :=
+  begin
+    dsimp [has_one.one, monoid.one],
+    exact continuous_const,
+  end,
+  mul_mem := λ f g fc gc, continuous.comp (continuous.prod_mk fc gc) (topological_monoid.continuous_mul β) }.
+
+instance continuous_subring (α : Type u) (β : Type v) [topological_space α] [topological_space β]
```

Comment:
I don't understand what is going on with the `dsimp`s. I do think that you should maybe split this instance in `continuous_add_submonoid` and `continuous_add_subgroup`.

Reply:
Done.